server/xpub-server/entities/manuscript/resolvers/index.js

```diff
@@ -80,6 +86,15 @@ const resolvers = {
     },
   },
 
+  Subscription: {
+    manuscriptUploadProgress: {
+      subscribe: async (_, vars, context) => {
+        const pubsub = await getPubsub()
+        return pubsub.asyncIterator(`${ON_UPLOAD_PROGRESS}.${vars.id}`)
```

Comment:
For context: This is where we're changing the subscriptions 'listener' key to use the manuscript id. Heres the code from `pubsweet`s version of the `Subscription` resolvers.

`node_modules/pubsweet-server/src/graphql/definitions/upload.js`

```
 Subscription: {
    uploadProgress: {
      subscribe: async (_, vars, context) => {
        const pubsub = await getPubsub()
        return pubsub.asyncIterator(`${ON_UPLOAD_PROGRESS}.${context.user}`)
      },
    },
  },
```

Unfortunately I couldn't find any way to override graphql resolvers so I couldn't redefine `uploadProgress`, instead I've had to create a new resolver called `manuscriptUploadProgress` which is then added to the server and clients graphql  schemas